packages/cli/src/models/network/NetworkController.ts

```diff
@@ -515,6 +558,7 @@ export default class NetworkController {
 
   // Proxy model
   public async upgradeProxies(packageName: string, contractAlias: string, proxyAddress: string, initMethod: string, initArgs: string[]): Promise<ProxyInterface[]> {
+    if (!isNotMigratableZosversion(this.currentZosversion)) await this.migrate();
```

Comment:
Nit: we can extract the if statement to a private method `await _migrateIfNeeded()` or so